Integrate Cancel Sending Message

Tanvi-Gaonkhadkar opened the pull request:
### Summary

This PR adds support for cancelling an in-flight chat message request in the frontend.

Closes #16

### Changes
- Added Cancel button while a message is being processed
- Used AbortController to cancel fetch requests
- Handled AbortError gracefully in the UI
- Ensured loading state is correctly reset on cancel

### Testing done
- Manually tested the Cancel button:
  - Clicking “Cancel” stops the message from being sent
  - Loading state resets correctly
  - AbortError is handled gracefully
- Tested with mock API for frontend development (React part only)
- Verified that other chat functionality works as before


### Notes
- Backend `/api/chat` exists and is fully functional, but this issue focused **only on frontend logic**
- Backend integration continues to work without changes
- The mock API was used just for frontend testing

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (`cancel-send-message`)
- [x] Ensure that the pull request title clearly represents the frontend feature implemented
- [x] Clearly described the frontend changes made (Cancel button, AbortController, loading state)
- [x] Linked to the relevant issue in GitHub (#16)
- [x] No related upstream or downstream pull requests
- [x] Demonstrated the feature works via manual frontend testing using mock API

Tanvi-Gaonkhadkar commented:
Hi,
I’ve removed all mock API logic and reverted to the real API calls.
Frontend tests, ESLint, and the production build are all passing locally.
Please let me know if any further changes are needed.


On Mon, Dec 22, 2025 at 2:54 PM Bervianto Leo Pratama <
***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In frontend/src/api/chatbot.ts
> <https://github.com/jenkinsci/resources-ai-chatbot-plugin/pull/64#discussion_r2639224363>
> :
>
> > @@ -1,6 +1,7 @@
>  import { type Message } from "../model/Message";
> -import { getChatbotText } from "../data/chatbotTexts";
> +// import { getChatbotText } from "../data/chatbotTexts";
>
> Yes, please.
>
> Thank you for contributing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/resources-ai-chatbot-plugin/pull/64#discussion_r2639224363>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BPUH3QLC7WAOI5UQF6K6AZ34C62FVAVCNFSM6AAAAACPXAW4HGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTMMBTGMYDGNZTG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

berviantoleo commented on frontend/src/components/Chatbot.tsx:
```diff
@@ -95,6 +99,7 @@ export const Chatbot = () => {
    * Handles the creation process of a chat session.
    */
   const handleNewChat = async () => {
+    console.log("Creating new chat")
```
Please remove some console.log

berviantoleo commented on frontend/src/tests/api_chatbot.test.ts:
```diff
@@ -96,41 +106,12 @@ describe("chatbotApi", () => {
     });
   });
-
-  describe("deleteChatSession", () => {
```
Not sure why you remove this one

berviantoleo commented on frontend/index.html:
```diff
@@ -6,7 +6,7 @@
   </head>
   <body>
-    <div id="root"></div>
+    <div id="chatbot-root"></div>
```
I'm not sure why you need to change this.

Tanvi-Gaonkhadkar commented:
I’ve rebased the branch and addressed your feedback:

   -

   Added UI feedback when message sending is cancelled
   -

   Removed unreachable code


On Thu, Dec 25, 2025 at 3:15 PM Bervianto Leo Pratama <
***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Please re-review your code. Some codes are not reachable.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/resources-ai-chatbot-plugin/pull/64#pullrequestreview-3612080488>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BPUH3QKYNCKCBMUOSHUSJPL4DOW3JAVCNFSM6AAAAACPXAW4HGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTMMJSGA4DANBYHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>

berviantoleo commented:
Please resolve the conflicts

Tanvi-Gaonkhadkar commented:
I’ve rebased the branch onto the latest main and force-pushed the updated
history.
The conflicts are now resolved and the branch is up to date.

On Sun, Dec 28, 2025 at 9:27 AM Bervianto Leo Pratama <
***@***.***> wrote:

> *berviantoleo* left a comment (jenkinsci/resources-ai-chatbot-plugin#64)
> <https://github.com/jenkinsci/resources-ai-chatbot-plugin/pull/64#issuecomment-3694429383>
>
> Please resolve the conflicts
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkinsci/resources-ai-chatbot-plugin/pull/64#issuecomment-3694429383>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BPUH3QNDCBBKS4NOMZLKVQ34D5ILXAVCNFSM6AAAAACPXAW4HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMOJUGQZDSMZYGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

berviantoleo commented:
Unfortunately, it still conflicts on our side. We cannot merge this PR.

![Screenshot_20251228_210331_Chrome](https://github.com/user-attachments/assets/b2acfd6e-8389-46ee-b73d-e12d754828ed)
